walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,187 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> mandatoryAttendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    long meetingDuration = request.getDuration();
+
+    if (meetingDuration > TimeRange.WHOLE_DAY.duration()) {
+        return Collections.emptyList();
+    }
+
+    if (request.getAttendees().isEmpty() && request.getOptionalAttendees().isEmpty()) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+   
+    List<TimeRange> unavailableTimeRanges = getUnavailableTimeRanges(events, mandatoryAttendees);
```

Comment:
Consider renaming this to unavailableMandatoryTimeRanges for clarity

Reply:
Done.